### PR TITLE
redux localstorage persistence, namespace filter/sort, spinners & detail view styling

### PR DESCRIPTION
**Pull request by ibolton336**

### CODE REVIEW

File: src/app/home/HomeComponent.tsx [hunk 1]
```diff
@@ -56,8 +56,10 @@ class HomeComponent extends React.Component<IProps, IState> {
   };
 
   componentDidMount() {
-    this.props.fetchClusters();
-    this.props.fetchStorage();
+    if (this.props.clusterList.length < 1) {
```
Comment (ibolton336) [inline src/app/home/HomeComponent.tsx]: temporary for ui work

**Comment by ibolton336** [comment]
closes #47

File: src/app/home/HomeComponent.tsx [hunk 2]
```diff
@@ -56,8 +59,10 @@ class HomeComponent extends React.Component<IProps, IState> {
   };
 
   componentDidMount() {
+    // if (this.props.clusterList.length < 1) {
```
Comment (bond95) [inline src/app/home/HomeComponent.tsx]: Please, remove this comments.

File: src/app/home/components/ClusterStatusIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+interface IProps {
+  isSuccessful: boolean;
+}
+
+const ClusterStatusIcon: React.FunctionComponent<IProps> = ({
+  isSuccessful,
+  ...rest
+}) => {
+  const SuccessIcon = styled(CheckIcon)`
```
Comment (bond95) [inline src/app/home/components/ClusterStatusIcon.tsx]: I don't like mixing CSS and JavaScript, but this is for another issue.

File: src/app/home/components/DataListComponent.tsx [hunk 4]
```diff
@@ -21,23 +23,30 @@ const DataListComponent = ({ dataList, type, ...props }) => {
                 aria-labelledby="simple-item1"
               >
                 <DataListCell width={1}>
+                  <ClusterStatusIcon isSuccessful={true} />
```
Comment (bond95) [inline src/app/home/components/DataListComponent.tsx]: You can write it like that:
```JSX
<ClusterStatusIcon isSuccessful />
```

File: src/app/plan/components/MigSourceForm.tsx [hunk 5]
```diff
@@ -50,6 +50,7 @@ class MigSourceForm extends React.Component<any> {
         <NamespaceTable
           setFieldValue={setFieldValue}
```
Comment (bond95) [inline src/app/plan/components/MigSourceForm.tsx]: As I see `setFieldValue` is already in `this.props`, so you can remove it here.

File: src/app/plan/duck/operations.ts [hunk 6]
```diff
@@ -64,13 +65,23 @@ const fetchPlan = () => {
...
-      dispatch(sourceClusterNamespacesFetchSuccess(res.data.items));
-    }).catch(err => AlertCreators.alertError('Failed to load namespaces for cluster'));
+    const client: IClusterClient = ClientFactory.forCluster(
+      clusterName,
+      getState(),
+    );
+    const nsResource = new CoreClusterResource(
+      CoreClusterResourceKind.Namespace,
+    );
+    client
```
Comment (bond95) [inline src/app/plan/duck/operations.ts]: Please, use async/await

File: src/helpers.js [hunk 7]
```diff
@@ -1,3 +1,3 @@
-import { createBrowserHistory } from 'history';
+import { createBrowserHistory } from 'history'
 
-export const history = createBrowserHistory();
```
Comment (bond95) [inline src/helpers.js]: Why do you removed here semicolons? Because as I see, other exports and imports has semicolons.

**Comment by bond95** [comment]
I think it would be nice squash commits before merge.

**Comment by andyvesel** [comment]
@ibolton336 needs rebase.
Also I'd suggest to rebase/squash all linting commits into one

File: package.json [hunk 8]
```diff
@@ -45,8 +45,10 @@
     "lodash": "4.17.11",
     "node-sass": "4.11.0",
     "prettier": "1.16.4",
+    "react-loader-spinner": "2.3.0",
     "react-select": "2.4.2",
     "react-table": "6.9.2",
+    "redux-persist": "^5.10.0",
```
Comment (bond95) [inline package.json]: Remove that sign `^` here too.

File: package.json [hunk 9]
```diff
@@ -77,6 +79,7 @@
     "emotion-theming": "10.0.9",
     "history": "4.9.0",
     "jsonwebtoken": "8.5.1",
+    "match-sorter": "^2.3.0",
```
Comment (bond95) [inline package.json]: Same

File: src/app/plan/duck/operations.ts (hunk 6 above)
Comment (bond95) [reply src/app/plan/duck/operations.ts]: Is there any reason, not use here async/await?

Comment (ibolton336) [reply src/app/plan/duck/operations.ts]: this is not related to this PR. Please open an issue. The only changes made here were from a linting tool.

**Comment by eriknelson** [comment]
@bond95 @andyvesel The repo is set to squash and merge by default when done from the Github UI

**Comment by bond95** [comment]
> @bond95 @andyvesel The repo is set to squash and merge by default when done from the Github UI

Ok, thank you :)

File: package.json (hunk 8 above)
Comment (ibolton336) [reply package.json]: create an issue to refactor dependencies.

Comment (bond95) [reply package.json]: As I see all dependencies are without that sign `^`, so leaving it is not consistent.

File: src/app/plan/duck/operations.ts (hunk 6 above)
Comment (bond95) [reply src/app/plan/duck/operations.ts]: Ok

File: package.json (hunk 8 above)
Comment (eriknelson) [reply package.json]: A bot updates this.

Comment (bond95) [reply package.json]: I see, thank you for your comment.

**Pull request merged**